sui/manage-coins.js

```diff
@@ -0,0 +1,218 @@
+const { TransactionBlock } = require('@mysten/sui.js/transactions');
+const { saveConfig } = require('../evm/utils');
+const { Command } = require('commander');
+const { addBaseOptions } = require('./cli-utils');
+const { getWallet } = require('./sign-utils');
+const { printInfo } = require('../evm/utils');
+const chalk = require('chalk');
+const { loadSuiConfig } = require('./utils');
+
+const SUI_COIN_ID = '0x2::sui::SUI';
+
+class CoinManager {
+
+    static async getAllCoins(client, account) {
+        let cursor;
+        const coinTypeToCoins = {};
+
+        do {
+            const coinsAtCursor = await client.getAllCoins({
+                owner: account,
+                limit: 100,
+                cursor,
+            });
+
+            for (const coin of coinsAtCursor.data) {
+                if (!coinTypeToCoins[coin.coinType]) {
+                    coinTypeToCoins[coin.coinType] = {
+                        data: [],
+                        totalBalance: 0n,
+                    };
+                }
+
+                coinTypeToCoins[coin.coinType].data.push(coin);
+                coinTypeToCoins[coin.coinType].totalBalance += BigInt(coin.balance);
+            }
+
+            if (coinsAtCursor.hasNextPage) {
+                cursor = coinsAtCursor.nextCursor;
+            }
+        } while (cursor);
+
+        return coinTypeToCoins;
+    }
+
+    static async splitCoins(tx, coinTypeToCoins, options) {
+        CoinManager.checkSplitAmount(options);
+        const splitAmount = BigInt(options.split);
+
+        console.log('\n==== Splitting Coins ====');
+
+        // Set coin type to given coin type or the first coin type if there's only one if it's a SUI token.
+        const hasOnlyGasToken = Object.keys(coinTypeToCoins).length === 1 && coinTypeToCoins[SUI_COIN_ID];
```

Comment:
split should only split for one coin type i think. If coin type isn't provided, Sui is assumed